From 8d63cc6f907a55599f2861e860fdf4c2f0e71bee Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:49 -0400
Subject: [scsi] hpsa: change doorbell reset delay to ten seconds

Message-id: <1410186734-19986-59-git-send-email-thenzl@redhat.com>
Patchwork-id: 92058
O-Subject: [RHEL6.6 e-stor PATCH 58/83] hpsa: change doorbell reset delay to ten seconds
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

After 3.22 firmware, PMC firmware guys tell us the
previous 5 second delay after a reset now needs to
be 10 secs to avoid a PCIe error due to the driver
looking at the controller too soon after the reset.

Signed-off-by: Justin Lindley <justin.lindley@hp.com>
Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Reviewed-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 00701a96f81f4e1b7279f5e05fe02274bbe29e59)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 022b59c..b142b5e 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -5844,12 +5844,12 @@ static int hpsa_controller_hard_reset(struct pci_dev *pdev,
   dev_info(&pdev->dev, "using doorbell to reset controller\n");
   writel(use_doorbell, vaddr + SA5_DOORBELL);
 
-  /* PMC hardware guys tell us we need a 5 second delay after
+  /* PMC hardware guys tell us we need a 10 second delay after
    * doorbell reset and before any attempt to talk to the board
    * at all to ensure that this actually works and doesn't fall
    * over in some weird corner cases.
    */
-  msleep(5000);
+  msleep(10000);
  } else { /* Try to do it the PCI power state way */
 
   /* Quoting from the Open CISS Specification: "The Power
-- 
1.7.1